xcode/Subconscious/Shared/Services/NoosphereService.swift

```diff
@@ -0,0 +1,201 @@
+//
+//  NoosphereService.swift
+//  Subconscious (iOS)
+//
+//  Created by Gordon Brander on 1/10/23.
+//
+//  Noosphere lifecycle:
+//  1. Initialize namespace (once ever)
+//  2. Create key (once per key)
+//  3. Create sphere for key (once per key)
+//  4. Retreive sphere identity from sphere creation receipt, then store (once, upon sphere creation)
+//  5. Retrieve sphere mnemonic from sphere creation receipt. Display to user, then discard (once, upon sphere creation).
+//  6. Open sphere file system (on-demand)
+
+import Foundation
+import SwiftNoosphere
+
+public enum NoosphereError: Error {
+    case foreignError(String)
+    case memoDoesNotExist
+}
+
+public struct SphereReceipt {
+    var identity: String
+    var mnemonic: String
+}
+
+struct SphereMemo: Hashable {
+    var contentType: String
+    var data: Data
+}
+
+/// Create a Noosphere instance.
+///
+/// - Property noosphere: pointer that holds all the internal book keeping.
+///   DB pointers, key storage interfaces, active HTTP clients etc.
+public final class Noosphere {
```

Comment:
Noosphere manages the lifetime of the Noosphere pointer, and provides methods related to Noosphere.